joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -148,29 +148,33 @@ controller_interface::return_type JointTrajectoryController::update(
   {
     bool first_sample = false;
     // if sampling the first time, set the point before you sample
-    if (!traj_external_point_ptr_->is_sampled_already())
+    if (!current_trajectory_->is_sampled_already())
     {
       first_sample = true;
       if (params_.open_loop_control)
       {
-        traj_external_point_ptr_->set_point_before_trajectory_msg(
-          time, last_commanded_state_, joints_angle_wraparound_);
+        if (fabs(last_commanded_time_.seconds()) < std::numeric_limits<float>::epsilon())
+        {
+          last_commanded_time_ = time;
+        }
+        current_trajectory_->set_point_before_trajectory_msg(
+          last_commanded_time_, last_commanded_state_, joints_angle_wraparound_);
```

Comment:
In this case, the `last_command_time_` would be same for the first 2 iterations. Is that okay?. If its fine, then all good.